backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/diagram/ViewDiagramDescriptionConverter.java

```diff
@@ -430,6 +435,29 @@ private Function<VariableManager, List<?>> getSemanticElementsProvider(org.eclip
         };
     }
 
+    private boolean isUnsynchronizedNodeDescription(org.eclipse.sirius.components.view.DiagramElementDescription elementDescription) {
+        return elementDescription instanceof org.eclipse.sirius.components.view.NodeDescription
+                && elementDescription.getSynchronizationPolicy().getName().equals(SynchronizationPolicy.UNSYNCHRONIZED.toString());
+    }
+
+    private List<Object> getUnsynchronizedSemanticElements(VariableManager variableManager, org.eclipse.sirius.components.view.NodeDescription nodeDescription) {
+        Object semanticElementIdsVariableContent = variableManager.getVariables().get(NodeComponent.SEMANTIC_ELEMENT_IDS);
+        if (semanticElementIdsVariableContent instanceof List<?>) {
+            IEditingContext editingContext = variableManager.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class).get();
```

Comment:
Do not call directly `.get` on an optional. We are explicitly using optionals to indicate that it can be null and thus throw a `NullPointerException`. You need to handle the case where this is an empty optional.

I'll do it myself in order to move forward with the review.